receiver/googlepubsubpushreceiver/README.md

```diff
@@ -0,0 +1,46 @@
+| Code coverage | [![codecov](https://codecov.io/github/open-telemetry/opentelemetry-collector-contrib/graph/main/badge.svg?component=receiver_googlepubsubpush)](https://app.codecov.io/gh/open-telemetry/opentelemetry-collector-contrib/tree/main/?components%5B0%5D=receiver_googlepubsubpush&displayType=list) |
+| [Code Owners](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#becoming-a-code-owner)    | [@axw](https://www.github.com/axw), [@constanca-m](https://www.github.com/constanca-m) |
+
+[development]: https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/component-stability.md#development
+<!-- end autogenerated section -->
+
+> **NOTE**: Refer to _[Difference to the current google pub sub receiver](#difference-to-the-current-google-pub-sub-receiver)_ if you don't know which receiver to use or just want to understand the differences.
+
+The `googlepubsubpush` receiver ingests OpenTelemetry data through Google Cloud Pub/Sub push subscriptions, supporting two primary data ingestion patterns:
+- Direct Log Ingestion: Processes Pub/Sub messages containing log data directly.
+ - GCS Event Processing: Handles Pub/Sub notifications for new files in Google Cloud Storage.
+
+All received data is parsed into OpenTelemetry format using configurable encoding extensions, enabling flexible and near real-time telemetry collection from multiple Google Cloud sources.
+
+## Example configuration
+
+```yaml
+receivers:
+  googlepubsubpush:
+    endpoint: :8080
+    encoding: googlecloudlogentry_encoding
+
+extensions:
+  googlecloudlogentry_encoding:
+```
+
+## Difference to the current google pub sub receiver
+
+The main difference between the current [`googlecloudpubsub` receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/googlecloudpubsubreceiver) and the proposed new receiver lies in how they receive messages from Google Pub/Sub: pull vs. push subscriptions. The table below summarizes their differences:
+
+|    Component     |                  `googlecloudpubsub`                  |           `googlepubsubpush`  (this receiver)           |
+|:----------------:|:-----------------------------------------------------:|:-------------------------------------------------------:|
+| **Subscription** |                         Pull                          |                          Push                           |
+|  **Data Flow**   | The receiver _polls_ Pub/Sub to ask for new messages. | Pub/Sub actively *pushes* new messages to the receiver. |
+|  **Initiator**   |                       Receiver                        |                         Pub/Sub                         |
+|    **Result**    | Potential latency between publication and collection. |         Near real-time, event-driven ingestion.         |
```

Comment:
I'm not so sure about that - do you have any references that suggest pull subscription may be slower? I would be inclined to drop this, maybe mention that push is more suitable when deploying the collector in Cloud Run, and otherwise just let the reader refer to the Pub/Sub docs for the differences.

Reply:
:ok_hand: not sure I follow as there's no interval configuration for pulling - the client automatically handles flow control. But anyway, dropping sounds good. We can add more later if there's still confusion.